lib/dockerregistry/inventory.go

```diff
@@ -1631,10 +1631,18 @@ func (sc *SyncContext) ExecRequests(
 	// We have to use a WaitGroup, because even though we know beforehand the
 	// number of workers, we don't know the number of jobs.
 	wg := new(sync.WaitGroup)
+
+	errorEncountered := false
+
 	// Log any errors encountered.
 	go func() {
 		for reqRes := range requestResults {
 			if len(reqRes.Errors) > 0 {
+				if !errorEncountered {
```

Comment:
Please remove this optimization for now, unless we deem it necessary in the future.